### ISSUE TITLE
[google_maps_flutter] Expose an "onTilesLoaded" to signal when all tiles have been fully rendered.

### ISSUE THREAD

**Issue by FranRiadigos** [flutter/flutter#99610]
## Use case

The flutter plugin exposes an `onMapCreated()` function that states when the map is ready.
However, until all tiles are fully loaded it shows a grid with grey lines, and any effect or any interaction with the map seems useless until tiles are rendered.

These are the use cases I wish to accomplish:
- Ensure all tiles are rendered before taking a snapshot.
Running the method `takeSnapshot()` after `onMapCreated()` leads to an image with either, partial tiles rendered or none at all.
- Zoom-in to a position once the map is first loaded.
The zoom effect is kind of weird if done after `onMapCreated()` is fired, since it will zoom a grid of grey lines.

## Proposal

Each platform already contains a functionality to determine when tiles are fully rendered:
- Android - `GoogleMap.OnMapLoadedCallback`
- iOS - `GMSMapViewDelegate.mapViewDidFinishTileRendering`
- Web - `Map.tilesloaded`

However, AFAIK, there is no such functionality in the flutter plugin.

**Comment by maheshmnj** [comment]
Hi @FranRiadigos, Thanks for filing the issue. Google maps plugin currently doesn't have such a feature to notify when the tiles are loaded that looks like a valid proposal.

**Comment by FranRiadigos** [comment]
Hi,

I recently got an issue on `Crashlytics` that might be related to my request.

Please, let me know whether I should open a bug issue about it.

When `onMapCreated` is fired, I call 
```dart 
controller.animateCamera(CameraUpdate.newLatLngBounds(bounds, padding));
```

Which produces the following error:

```
com.google.maps.api.android.lib6.common.apiexception.c: Error using newLatLngBounds(LatLngBounds, int): Map size can't be 0. Most likely, layout has not yet occured for the map view. Either wait until layout has occurred or use newLatLngBounds(LatLngBounds, int, int, int) which allows you to specify the map's dimensions. at com.google.maps.api.android.lib6.common.m.i(:com.google.android.gms.dynamite_mapsdynamite@221514065@22.15.14 (100400-0):0) ... 
```

The `GoogleMap` widget is rendered using hybrid composition.

``` dart
AndroidGoogleMapsFlutter.useAndroidViewSurface = true;
```

Thanks.

**Comment by maheshmnj** [comment]
@FranRiadigos, Please file that as a separate issue with all the requested details in the issue template and we will take a look at it.

**Comment by jhonatan-3a** [comment]
This feature would be really appreaciated, im also working on something that would be useful to know whether or not the map tiles were loaded

**Comment by oelburk** [comment]
Yeah, would be really nice! Any update on when this will be planned? Thanks!

**Comment by singh3abhi** [comment]
Hey, any updates on this issue ?

**Comment by Firdavs-coder** [comment]
hey, any updates ?

**Comment by mariomurrent-softwaresolutions** [comment]
That would be really nice and helpful. Any updates on this?
